scripts/symbolize.py

```diff
@@ -128,13 +137,101 @@ def resolve(self, addr):
             ret = '!!!'
         return ret
 
+    def symbol_plus_offset(self, addr):
+        ret = None
+        prevsize = 0
+        reladdr = self.subtract_load_addr(addr)
+        elf = self.get_elf(self._bin)
+        cmd = self.arch_prefix('nm')
+        if not reladdr or not elf or not cmd:
+            return ''
+        ireladdr = int(reladdr, 0)
+        nm = subprocess.Popen([cmd, '--numeric-sort', '--print-size', elf],
+                               stdin = subprocess.PIPE,
+                               stdout = subprocess.PIPE)
+        for line in iter(nm.stdout.readline, ''):
+            try:
+                addr, size, _, name = line.split()
+            except:
+                # Size is missing
+                addr, _, name = line.split()
+                size = '0'
+            iaddr = int(addr, 16)
+            isize = int(size, 16)
+            if iaddr == ireladdr:
+                ret = name
+                break
+            if iaddr < ireladdr and iaddr + isize >= ireladdr:
+                offs = ireladdr - iaddr
+                ret = name + '+' + str(offs)
+                break
+            if iaddr > ireladdr and prevsize == 0:
+                offs = iaddr + ireladdr
+                ret = prevname + '+' + str(offs)
+                break
+            prevsize = size
+            prevname = name
+        if ret is None:
```

Comment:
duplication of the same check`if ret is None`. I suggest to adda function like
```
def xstr(s):
    if s is None:
        return ''
    return str(s)
```
and then wrap , for example on lines 220, 221 with this function call:
```
if sym:
    ret += ' ' + xstr(sym)
```

Reply:
I'll fix as above (initialize `ret = ''`).